views/dashboard/about-page/components/FormView.tsx

```diff
@@ -0,0 +1,221 @@
+import { useState, useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+import useForm from '@/utils/hooks/useForm'
+import FormValidations from '@/utils/validations/forms'
+import { FileUpload, FormButtonGroup, TextField } from '@/components/form'
+import { updateAboutPageData } from '@/lib/db/cms/about-page'
+import { setAbout } from '@/lib/redux/slices/aboutPage'
+import { StyledForm } from './StyledFormView'
+import { handleCloudUpload } from '@/utils/fileConversions'
+import { enqueueSnackbar } from 'notistack'
+import { statuses, messages } from '@/utils/banners'
+
+const initialState = {
+    name: { value: '', error: '' },
+    description: { value: '', error: '' },
+    header: { value: '', error: '' },
+    subHeader: {
+        value: '',
+        error: ''
+    },
+    img: { value: [], error: '' },
+
+    role: {
+        value: '',
+        error: ''
+    },
+    title: {
+        value: '',
+        error: ''
+    }
+}
+
+interface Payload {
+    [key: string]: any
+}
+
+const validations = {
+    name: FormValidations.yolo,
+    description: FormValidations.yolo,
+    header: FormValidations.yolo,
+    subHeader: FormValidations.yolo,
+    img: FormValidations.yolo,
+    role: FormValidations.yolo,
+    title: FormValidations.yolo
+}
+
+const FormView = ({ store }: any) => {
+    const dispatch = useDispatch()
+
+    const { description, header, subHeader, img, role, title, name, id } = store
+
+    const {
+        disableSubmit,
+        form,
+        formLoading,
+        handleChange,
+        handleFormSubmit,
+        handleNonFormEventChange,
+        handleImport,
+        isDataImported
+    } = useForm(initialState, validations, store)
+
+    const onFormSubmit = async () => {
+        if (!store) {
+            return
+        }
+        if (!form) {
+            return
+        }
+
+        try {
+            const payload: Payload = {}
+            for (const [key] of Object.entries(store)) {
```

Comment:
Swap `store` with `form`. You're doing it backwards here. You want to loop over the form cuz the form fields are the relevant data set, store has extra things like id, that aren't relevant to the actual payload